### PR TITLE
fix: update without color on superset of nodecollection resets color

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
Basically replicates https://github.com/cognitedata/reveal/pull/1816 for the 3.0 alpha branch.

According to Christopher, this should not contribute too much to the pain it is to rebase.

### CODE REVIEW

**Review by larsmoa — COMMENTED** [review]
Update title to clearly reflect what issue is fixed so generating change log on release is easier.

File: viewer/packages/rendering/src/rendering/NodeAppearanceTextureBuilder.ts [hunk 1]
```diff
@@ -234,7 +234,10 @@ function fillRGBA(rgbaBuffer: Uint8ClampedArray, style: NodeAppearance) {
 function combineRGBA(rgbaBuffer: Uint8ClampedArray, treeIndices: IndexSet, style: NodeAppearance) {
   const [r, g, b, a] = appearanceToColorOverride(style);
   // Create a bit mask for updating color (update if style contains color, don't update if it doesn't)
-  const updateRgbBitmask = style.color !== undefined ? 0b11111111 : 0;
+  const updateRgbBitmask =
+    style.color !== undefined && !(style.color[0] === 0 && style.color[1] === 0 && style.color[2] === 0)
```
Comment (pramodcog) [inline viewer/packages/rendering/src/rendering/NodeAppearanceTextureBuilder.ts:238]: `!(style.color[0] === 0 && style.color[1] === 0 && style.color[2] === 0)` can be changed to `!style.color.every(value => value === 0)`

**Review by pramodcog — CHANGES_REQUESTED** [review]

File: viewer/packages/rendering/src/rendering/NodeAppearanceTextureBuilder.ts (hunk 1 above)
Comment (haakonflatval-cognite) [reply viewer/packages/rendering/src/rendering/NodeAppearanceTextureBuilder.ts]: That looks better. Thanks

**Review by pramodcog — APPROVED** [review]
Looks good to me

**Pull request merged**